### PR TITLE
Replace plainbox references (Infra)

### PR DESCRIPTION
**Pull request by waveform80**
## Description

Replaces references to plainbox with checkbox or, where sensible, simply removes them. This only affects files under `docs/` but even then there are places where plainbox references remain, such as in category and job identifiers. However, I'm reasonably confident that all the changes proposed are within prose only.

There are some references I've simply removed, because they reference specific plainbox versions and attempting to map these to checkbox versions is difficult and probably pointless (as they'll apply to all checkbox versions in current usage). There are also some "other" fixups, like a couple of obvious typos, a corrected path, and some fixed links.

The changes are organized into separate commits with the trivial obvious stuff first (where I'm confident that replacement of plainbox with checkbox is all that's needed), and the later commits covering stuff I'm a little less sure about which should probably be double-checked.

Each commit has additional commentary in the commit message. Where long paragraphs have been changed, I've re-wrapped things. If this makes things too hard to review, I'm happy to separate the re-wrap into a separate commit or just avoid the re-wrap entirely (and to rebase or re-work generally).

## Resolved issues

Fixes: #1278

## Tests

Tested build with sphinx locally; changes look reasonable and well formatted.

### CODE REVIEW

File: docs/how-to/custom-app.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ Let's create a Checkbox Snap for project called "myproject".
 
 .. code-block:: bash
 
-    $ ./configure --with-provider-included project=myproject base=18 && ./cleanup.sh
+    $ ./configure --with-provider-included project=myproject base=18
```
Comment (Hook25) [inline docs/how-to/custom-app.rst:31]: Why are you removing the cleanup command? also, probably unrelated to this

**Comment by Hook25** [comment]
/canonical/self-hosted-runners/run-workflows 9c8760f60ae5bff5f8ebeedc2facf24260f19138

**Review by Hook25 — CHANGES_REQUESTED** [review]
I have to ask you just one change on this, please make all the `checkbox` -> `Checkbox` (I would do it myself but I can't push to your fork!)

Ty for this btw, really helps us out

**Comment by Hook25** [comment]
/canonical/self-hosted-runners/run-workflows 3cb4e459788c6d5cbe534f09cfa3b5f41dd624fb

**Review by Hook25 — APPROVED** [review]
Thank you very much for taking this up!

+1

**Comment by Hook25** [comment]
In order to land this we need signed commits, can you please sign your commits?

https://github.com/canonical/checkbox/blob/main/CONTRIBUTING.md#pull-requests

**Comment by waveform80** [comment]
Should all be signed now (and verified after realizing the GPG key in GitHub hadn't been updated with new emails in forever)

**Comment by Hook25** [comment]
/canonical/self-hosted-runners/run-workflows fc197c3ce4ec5b9b7358ab9a5f259c94fe8f577f

**Pull request merged**